.scripts/aoe4quicksheet/sync.ts

```diff
@@ -92,29 +96,31 @@ function mapSheetUnitToStandardFormat(unitData: MappedSheetUnit): StandardUnitFo
       time: +unitData.buildTime,
     },
 
-    attack: {
-      melee: unitData.attackType == "Melee" ? +unitData.totalAttack : 0,
-      ranged: unitData.attackType == "Ranged" ? +unitData.totalAttack : 0,
-      siege: unitData.attackType == "True" ? +unitData.totalAttack : 0,
-      fire: +unitData.torchAttack,
-      speed: round(+unitData.attackSpeed, 2),
-      dps: round(+unitData.damagePerSecond, 1),
-    },
-
-    range: {
-      min: round(+unitData.minRange, 1),
-      max: round(+unitData.maxRange, 1),
-    },
-
-    vision: {
-      lineOfSight: +unitData.lineOfSight,
-      heightOfSight: +unitData.heightOfSight,
-    },
-
-    armor: {
-      melee: +unitData.meleeArmor,
-      ranged: +unitData.rangedArmor,
-      fire: +unitData.fireArmor,
+    weapons: +unitData.totalAttack
+      ? [
+          {
+            type: attackTypeMap[unitData.attackType as string],
+            damage: +unitData.totalAttack,
+            speed: +unitData.attackSpeed,
+            range: +unitData.maxRange
+              ? {
+                  min: round(+unitData.minRange, 1),
+                  max: round(+unitData.maxRange, 1),
+                }
+              : undefined,
+          },
+        ]
+      : [],
+
+    armor: [
+      +unitData.rangedArmor && { type: "ranged", value: +unitData.rangedArmor },
```

Comment:
had no idea about `+` in TS, cool

Reply:
Hah I discovered this only yesterday. It's [actually vanilla JS ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus), in TS it is the most concise way to cast a `string | number` to  t a number.